src/django/api/views/v1/opensearch_query_builder/opensearch_query_builder.py

```diff
@@ -123,26 +99,41 @@ def __build_os_id(self, values):
             }
         )
 
+    @abstractmethod
+    def _build_date_range(self, query_params):
```

Comment:
Why this method should have the @abstractmethod decorator, as it is only used in the ModerationEventsQueryBuilder?
In my opinion, the definition of the _build_date_range method should be moved into the OpenSearchQueryBuilder and it should be marked as private.

Reply:
> In my opinion, the definition of the _build_date_range method should be moved into the OpenSearchQueryBuilder and it should be marked as private.

I agree with you. Fixed. Thank you for pointing out this.